test/kube_checks.go

```diff
@@ -75,6 +75,55 @@ func WaitForPodListState(client *KubeClient, inState func(p *corev1.PodList) (bo
 	})
 }
 
+// WaitForPodState polls the status of the specified Pod
+// from client every interval until inState returns `true` indicating it
+// is done, returns an error or timeout. desc will be used to name the metric
+// that is emitted to track how long it took to get into the state checked by inState.
+func WaitForPodState(client *KubeClient, inState func(p *corev1.Pod) (bool, error), name string, namespace string) error {
+	p := client.Kube.CoreV1().Pods(namespace)
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForPodState/%s", name))
+	defer span.End()
+
+	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
+		p, err := p.Get(name, metav1.GetOptions{})
+		if err != nil {
+			return true, err
+		}
+		return inState(p)
+	})
+}
+
+// WaitForServiceHasAtLeastOneEndpoint polls the status of the specified Service
+// from client every interval until number of service endpoints = numOfEndpoints
+func WaitForServiceEndpoints(client *KubeClient, svcName string, svcNamespace string, numOfEndpoints int) error {
+	endpointsService := client.Kube.CoreV1().Endpoints(svcNamespace)
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForServiceHasAtLeastOneEndpoint/%s", svcName))
+	defer span.End()
+
+	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
+		endpoints, err := endpointsService.List(metav1.ListOptions{})
+		if err != nil {
+			return true, err
+		}
+
+		for _, e := range endpoints.Items {
+			if e.Name == svcName && countEndpointsNum(&e) == numOfEndpoints {
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+}
+
+func countEndpointsNum(e *corev1.Endpoints) int {
```

Comment:
I am pretty sure we have such a method somewhere, but it might be in serving only.

Reply:
I didn't found such function in pkg, I tried to seach a bit without results